src/components/EnhancedComponent/EnhancedComponent.js

```diff
@@ -50,38 +61,35 @@ class EnhancedComponent extends Component {
   }
 
   render() {
-    return <Provider store={this.store}>{this.renderComponent()}</Provider>
+    return (
+      <Provider store={this.store}>
+        {this.renderStateComponent()}
+      </Provider>
+    )
   }
 
-  getPropsForState() {
-    const {data, selector} = this.props
-    return transformDataToProps(data, selector)
-  }
-
-  updatePropsInState() {
-    if (this.shouldUpdateState()) {
-      const props = this.getPropsForState()
-      this.setState({
-        ...props,
-      })
-    }
+  updateState() {
+    const props = transformDataToProps(this.props.component.props)
+    this.setState({
+      ...props,
+    })
   }
 
-  unwatch() {
-    this.collections.forEach(collection => {
-      collection.off(COLLECTION_EVENTS, this.updatePropsInState, this)
+  unwatchState() {
+    this.stateCollections.forEach(collection => {
+      collection.off(COLLECTION_EVENTS, this.updateState, this)
     })
-    this.models.forEach(model => {
-      model.off(MODEL_EVENTS, this.updatePropsInState, this)
+    this.stateModels.forEach(model => {
+      model.off(MODEL_EVENTS, this.updateState, this)
     })
   }
 
-  watch() {
-    this.collections.forEach(collection => {
-      collection.on(COLLECTION_EVENTS, this.updatePropsInState, this)
+  watchState() {
+    this.stateCollections.forEach(collection => {
+      collection.on(COLLECTION_EVENTS, this.updateState, this)
     })
-    this.models.forEach(model => {
-      model.on(MODEL_EVENTS, this.updatePropsInState, this)
+    this.stateModels.forEach(model => {

```

Comment:
These for `forEach` calls seem to be doing the same thing, there could a way to simplify it (or not, cause I'm just looking at the code and not behind the code ... if that makes sense :)).

Reply:
Updated!